turbopack/crates/turbopack-core/src/module_graph/mod.rs

```diff
@@ -347,6 +340,66 @@ impl SingleModuleGraph {
             }
         }
 
+        // If there were any
+        if !issues.is_empty() {
+            let roots_indices = entries
+                .iter()
+                .flat_map(|m| m.entries())
+                .map(|m| *modules.get(&m).expect("all roots should have been visited"))
+                .collect::<FxHashSet<_>>();
+            // Reverse the graph so we can find routes to roots.
+            let reversed_graph = Reversed(&graph);
+            for (module_idx, new_issues) in issues {
+                let Some((_, path)) = petgraph::algo::astar(
+                    reversed_graph,
+                    module_idx,
+                    |n| roots_indices.contains(&n),
+                    // Edge weights
+                    |e| match e.weight() {
+                        // Prefer following normal imports/requires when we can
+                        ChunkingType::Parallel { .. } => 0,
+                        _ => 1,
+                    },
+                    // `astar` can be accelerated with a distance estimation heuristic, however we
```

Comment:
You could compute the depth of each node in the `into_breadth_first_edges` loop above and use that as the heuristic here?